### ISSUE TITLE
API Adds Email Multiple Times Duplicated

### ISSUE THREAD

**Issue by rayliverified** [twisted1919/mailwizz-php-sdk#20]
The MailWizz API adds the same email multiple times. 
![maxthonsnap20171222132457](https://user-images.githubusercontent.com/19896174/34310234-ee44db28-e71b-11e7-8bd0-a038f74e80ac.png)

This is my code for adding an email:
`        //MailWizz
        require_once dirname(__FILE__) . '/mailwizz.init.php';
        $endpoint = new MailWizzApi_Endpoint_ListSubscribers();
        //App New Users
        $response = $endpoint->create('wj730zl99w094', array(
            'EMAIL'    => $email
        ));`

Doesn't MailWizz automatically check if a subscriber is already added and if so, not add that subscriber again?

**Comment by twisted1919** [comment]
Do you have any email verification service enabled ?

**Comment by rayliverified** [comment]
No.

I did some more testing and I believe I have isolated the reason for the problem. 
I am calling the MailWizz API in a POST request and for some reason, the POST request is executed multiple times. The POST request is only called once, but the server executes it multiple times which is why the MailWizz API is adding the same email multiple times. When the MailWizz request is removed from the POST request, the POST request only executes once. 

I do not know why MailWizz would cause a POST request to repeat. This seems like a separate issue. 

The issue here is that MailWizz adds an email even when it already exists. 

My post request:
```
<?php
if (!empty($_POST)) {

    $email = isset($_POST['email']) ? $_POST['email'] : '';

    $email = helper::clearText($email);
    $email = helper::escapeText($email);

    $result = array();
    $result = $helper->isEmailExistsInfo($email);

    if ($result['error'] !== false)
    {
        //MailWizz
        require_once dirname(__FILE__) . '/mailwizz.init.php';
        $endpoint = new MailWizzApi_Endpoint_ListSubscribers();
        //Delete user if exists in list so user can be added again.
        $endpoint->deleteByEmail('wj730zl99w074', $email);
        //App New Users
        $response = $endpoint->create('wj730zl99w074', array(
            'EMAIL'    => $email
        ));
    }

    echo json_encode($result);
    exit;
}

```

**Comment by twisted1919** [comment]
Maybe you should use `createUpdate` instead of `create` and don't call `deleteByEmail` anymore, so try that and see how it goes.

On another note, Mailwizz does check for dupes, but somehow your request is multiplied at the same time and therefore is too fast for mailwizz to catch it for some reason, i assume mainly because the database check for dupes takes too long. You might wanna look at https://kb.mailwizz.com/articles/performance-improvements/ or try to see why the post request gets multiplied, as mailwizz doesn't do this.

**Comment by rayliverified** [comment]
I did some more research and it looks like 90% of the issue is because my client is retrying the POST request due to timeout which is why the POST request is being called multiple times. However, the max number of retries is 1 which means at most, the email should only be added twice. 

There are instances where emails have been added 3 times. Could the internal HttpClient used by MailWizz have a retry timeout too? See this StackOverflow for a description of the default behavior of HttpClient: https://stackoverflow.com/questions/23054289/httpclient-executes-requests-multiple-time-if-request-timed-out

Thanks for the help!
